packages/mui-material/src/Input/Input.js

```diff
@@ -42,7 +42,7 @@ const InputRoot = styled(InputBaseRoot, {
     ];
   },
 })(({ theme, ownerState }) => {
-  const light = theme.palette.mode === 'light';
+  const light = (theme.vars || theme).palette.mode === 'light';
   const bottomLineColor = light ? 'rgba(0, 0, 0, 0.42)' : 'rgba(255, 255, 255, 0.7)';
```

Comment:
This goes against the purpose of using the new CSS variables structure, as we would still be depending on JS transformations. My proposal would be using the following:

```suggestion
  const light = (theme.vars || theme).palette.mode === 'light';
  const bottomLineColor = light ? 'rgba(0, 0, 0, 0.42)' : 'rgba(255, 255, 255, 0.7)';
  // TODO v6 update this value
  // const bottomLineColor = theme.vars ? theme.vars.palette.input.secondary : theme.palette.light ? 'rgba(0, 0, 0, 0.42)' : 'rgba(255, 255, 255, 0.7)';
```

The value for the light theme is a bit different (`rgba(0, 0, 0, 0.6)`), but it is the closest to what was previously used. cc @siriwatknp  

If we don't want this value to change, we can create a new prop in the palette, something like `inputBorderColor` (note that we can add this only in the `experimental_extendTheme` default theme.